### ISSUE TITLE
connection problems

### ISSUE THREAD

**Issue by marix11** [siacs/Conversations#1343]
Hi, after connection change or being long time without connection it doesnt connect to all my servers. Happens same on other phones in my family. Enabling disabling servers doesnt help. Only solution restart phone
Cant say from which version it started but never happened  in earlier versions. V1.5.2

**Comment by iNPUTmice** [comment]
What error message are you getting?

**Comment by marix11** [comment]
Failed connections and reconnecting endless like being offline

**Comment by Zibc** [comment]
I also suffer from connection issues since the last update (it seems)
my account is constantly "connecting" and turns to "Offline" after some time, although the server log says that the client is connected (server runs metronome/prosody)

**Comment by iNPUTmice** [comment]
please update to 1.6.1 and try to post an adb logcat output of the connection attempts.

**Comment by Zibc** [comment]
I'm a complete noob at using adb but I gave it a try, this thing spits out a lot of information (almost scary I must say) (help on cleaning the log is welcome!) (here is just a grep on 'conversation')

```
D/conversations( 8876): me@mydomain.tld/mobile: time out during connect reconnecting
D/conversations( 8876): me@mydomain.tld: Socket closed
D/conversations( 8876): me@mydomain.tld: connecting
D/conversations( 8876): using dns server: 192.168.0.254 to look up mydomain.tld
D/conversations( 8876): me@mydomain.tld: using values from dns bla.mydomain.tld[serverip]:5222
D/conversations( 8876): me@mydomain.tld: TLS connection established
D/conversations( 8876): me@mydomain.tld/mobile: Authenticating with SCRAM-SHA-1
D/conversations( 8876): me@mydomain.tld: logged in
D/conversations( 8876): me@mydomain.tld: resuming after stanza #23
D/conversations( 8876): me@mydomain.tld: session resumed
D/conversations( 8876): resending 0 iq stanza
D/conversations( 8876): me@mydomain.tld: sending intial ping
D/conversations( 8876): me@mydomain.tld: requesting ack for stanza #22
D/conversations( 8876): me@mydomain.tld: initial ping failed
D/conversations( 8876): me@mydomain.tld: acknowledging stanza #24
D/conversations( 8876): me@mydomain.tld: server acknowledged stanza #22
D/conversations( 8876): me@mydomain.tld/mobile: time out during connect reconnecting
D/conversations( 8876): me@mydomain.tld: Socket closed
D/conversations( 8876): me@mydomain.tld: connecting
D/conversations( 8876): using dns server: 192.168.0.254 to look up mydomain.tld
D/conversations( 8876): me@mydomain.tld: using values from dns bla.mydomain.tld[serverip]:5222
D/conversations( 8876): me@mydomain.tld: TLS connection established
D/conversations( 8876): me@mydomain.tld/mobile: Authenticating with SCRAM-SHA-1
D/conversations( 8876): me@mydomain.tld: logged in
D/conversations( 8876): me@mydomain.tld: resuming after stanza #24
D/conversations( 8876): me@mydomain.tld: session resumed
D/conversations( 8876): resending 0 iq stanza
D/conversations( 8876): me@mydomain.tld: sending intial ping
D/conversations( 8876): me@mydomain.tld: requesting ack for stanza #23
D/conversations( 8876): me@mydomain.tld: initial ping failed
D/conversations( 8876): me@mydomain.tld: acknowledging stanza #25
D/conversations( 8876): me@mydomain.tld: server acknowledged stanza #23
D/conversations( 8876): resetting all attepmt counts
D/conversations( 8876): me@mydomain.tld/mobile: time out during connect reconnecting
D/conversations( 8876): me@mydomain.tld: Socket closed
D/conversations( 8876): me@mydomain.tld: connecting
D/conversations( 8876): using dns server: 192.168.0.254 to look up mydomain.tld
D/conversations( 8876): me@mydomain.tld: using values from dns bla.mydomain.tld[serverip]:5222
D/conversations( 8876): me@mydomain.tld: TLS connection established
D/conversations( 8876): me@mydomain.tld/mobile: Authenticating with SCRAM-SHA-1
D/conversations( 8876): me@mydomain.tld: logged in
D/conversations( 8876): me@mydomain.tld: resuming after stanza #25
D/conversations( 8876): me@mydomain.tld: session resumed
D/conversations( 8876): resending 0 iq stanza
D/conversations( 8876): me@mydomain.tld: sending intial ping
D/conversations( 8876): me@mydomain.tld: requesting ack for stanza #24
D/conversations( 8876): me@mydomain.tld: initial ping failed
D/conversations( 8876): me@mydomain.tld: acknowledging stanza #26
D/conversations( 8876): me@mydomain.tld: server acknowledged stanza #24
D/conversations( 8876): resetting all attepmt counts
D/conversations( 8876): me@mydomain.tld/mobile: time out during connect reconnecting
D/conversations( 8876): me@mydomain.tld: Socket closed
```

**Comment by iNPUTmice** [comment]
@Zibc thanks a lot. Exactly what I was looking for.

What version are you on? PlayStore? or self compiled? (Just asking to figure out the best way to provide you with a possible fix.

**Comment by Zibc** [comment]
I'm using 1.6.1 from the play store.

**Comment by iNPUTmice** [comment]
@Zibc if you send me an email to inputmice@siacs.eu I will provide you with an APK that probably fixes that issue.

**Comment by Zibc** [comment]
I confirm that the 1.6.2 solves the problem I had.
Working great again!

Good job by the way and thanks for the quick reaction!

**Comment by marix11** [comment]
I will wait fdroid to test

**Comment by polytan02** [comment]
Same for me. Since 1.6.1 I cannot connect to my xmpp server.
Apk installed from the Play store.

**Comment by iNPUTmice** [comment]
the update is on its way to the playstore

**Comment by polytan02** [comment]
I've just had the update 1.6.3 and it solves the issue for me. I can now connect.

By the way, since the update the messages sent from others are in ugly green bubles. Would it be possible to revert to previous colours ? I don't mind the green as a colour but it is way too strong. A light pale green would be much nicer ! :)

**Comment by marix11** [comment]
Updated to 1.6.6. Still getting connection issues after connection changes. For example from home/car wifis. Its says connecting-offline all time and never reconnects

**Comment by iNPUTmice** [comment]
@matrix11 please try to connect a debugger the next time this happens (adb
logcat -s conversations)

**Comment by marix11** [comment]
Maybe you could icon in notification area if conversations online/offline. Since this error i see my contacts online but me.xan be offline for whole day

**Comment by marix11** [comment]
I have checked logcat. Timeout in dns errors and never connect

**Comment by marix11** [comment]
Seems fixed in 1.6.9. Will test more
